src/Data/Geometry/Structure/Feature.hs

```diff
@@ -99,6 +101,34 @@ sjLat (ScenarioJSON js) = getProp "lat" js
 sjAlt :: ScenarioJSON -> Maybe Float
 sjAlt (ScenarioJSON js) = getProp "alt" js
 
+sjBlockColor :: ScenarioJSON -> Maybe (JSString)
+sjBlockColor sj = getHexColor "defaultBlockColor" sj
+sjActiveColor :: ScenarioJSON -> Maybe (JSString)
+sjActiveColor sj = getHexColor "defaultActiveColor" sj
+sjStaticColor :: ScenarioJSON -> Maybe (JSString)
+sjStaticColor sj = getHexColor "defaultStaticColor" sj
+sjLineColor :: ScenarioJSON -> Maybe (JSString)
+sjLineColor sj = getHexColor "defaultLineColor" sj
+
+getHexColor :: JSString -> ScenarioJSON -> Maybe JSString
+getHexColor name = asLikeJS . js_getHexColor name
+
+foreign import javascript unsafe "if ($2.hasOwnProperty('properties') &&\
+                                 \    $2['properties'].hasOwnProperty($1) &&\
+                                 \    $2['properties'][$1].match(/^(#[A-Fa-f0-9]{6})$/)) {\
+                                 \$r = $2['properties'][$1];}"
```

Comment:
Need to add "else" variant to setup a proper result value. Otherwise, $r is undefined - which is very dangerous